pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -337,19 +337,22 @@ func updateTaskRunsStatus(pr *v1alpha1.PipelineRun, pipelineState []*resources.R
 	}
 }
 
-func (c *Reconciler) createTaskRun(logger *zap.SugaredLogger, rprt *resources.ResolvedPipelineRunTask, pr *v1alpha1.PipelineRun, sa, storageBasePath string, pipelineTimeout *metav1.Duration) (*v1alpha1.TaskRun, error) {
-	ts := rprt.ResolvedTaskResources.TaskSpec.DeepCopy()
-	if pipelineTimeout != nil {
-		pTimeoutTime := pr.Status.StartTime.Add(pipelineTimeout.Duration)
-		if ts.Timeout == nil || time.Now().Add(ts.Timeout.Duration).After(pTimeoutTime) {
+func (c *Reconciler) createTaskRun(logger *zap.SugaredLogger, rprt *resources.ResolvedPipelineRunTask, pr *v1alpha1.PipelineRun, sa, storageBasePath string) (*v1alpha1.TaskRun, error) {
+	var taskRunTimeout = &metav1.Duration{Duration: 0 * time.Second}
+	if pr.Spec.Timeout != nil {
+		pTimeoutTime := pr.Status.StartTime.Add(pr.Spec.Timeout.Duration)
+		if time.Now().After(pTimeoutTime) {
```

Comment:
Can we simplify the timeout logic?  I completely agree change is done in another PR. 

```
taskRunTimeout = &metav1.Duration{Duration: 0 * time.Second}
if pr.Spec.Timeout != nil {
    pTimeoutTime := pr.Status.StartTime.Add(pr.Spec.Timeout.Duration)
              // set taskrun timeout only if its valid timeout
          if !time.Now().After(pTimeoutTime) {
            taskRunTimeout = pr.Spec.Timeout
        }
    } else {
     // set timeout as nil if PR timeout is not set
        taskRunTimeout = nil
    }
```